nix/scripts/setup.sh

```diff
@@ -38,6 +40,33 @@ nix_install() {
         echo "Please see: https://nixos.org/nix/manual/#chap-installation" >&2
         exit 1
     fi
+
+    # Additional fixes
+    nix_create_user_dirs
+    nix_make_user_trusted
+    nix_daemon_restart
+}
+
+# Temporary fix for a known issue with Nix 2.14.
+# https://github.com/status-im/status-mobile/issues/16344
+nix_create_user_dirs() {
+    echo -e "Creating missing user profile and gcroot..." >&2
+    NIX_USER_PROFILE="/nix/var/nix/profiles/per-user/${USER}"
+    NIX_USER_GCROOT="/nix/var/nix/gcroots/per-user/${USER}"
+    sudo mkdir "${NIX_USER_PROFILE}" "${NIX_USER_GCROOT}"
+    sudo chown "${USER}" "${NIX_USER_PROFILE}" "${NIX_USER_GCROOT}"
+}
+
+# Fix for warnings about ignoring binary cache.
+nix_make_user_trusted() {
+    # Singl-user installations do not have this issue.
+    [[ ! -f /etc/nix/nix.conf ]] && return
+    echo -e "Adding ${USER} to trusted-users..." >&2
+    if grep trusted-users /etc/nix/nix.conf 2>/dev/null; then
+        sed -i "s/trusted-users = \(.*\)$/trusted-users = \1 $USER/" /etc/nix/nix.conf
+    else
+        echo "trusted-users = $USER" | sudo tee -a /etc/nix/nix.conf >/dev/null
```

Comment:
> Adding a user to trusted-users is essentially equivalent to giving that user root access to the system.
> -- https://nixos.org/manual/nix/stable/command-ref/conf-file.html#conf-trusted-users

I know you know, but honest question, is there an alternative?

Reply:
> The alternative is to open an issue with Nix and have them fix it. Because honestly I don't know why being a trusted user is a requirement to add a nix cache.

My guess is that it's all related to the fact that trusting a cache of binaries involves some known risks, so maybe they're pushing hard to make it more difficult for casual users (?)

> Also, honestly, I don't see a reason why not, as the current user already has root access through sudo on pretty much any developer machine, so I don't see how this is an issue.

I agree in practice it's fine, but I think ideally developers shouldn't need root access to write mobile apps, but maybe I'm idealizing things too much giving the insane number of dependencies in a RN project.

Anyways, thanks @jakubgs

Reply:
I agree, you should not need "root" to develop apps, but it is what it is. The one solution could be to allow people to pick which type of installation they want. For MacOS we are forced to use multi-user, but for linux single-user is still an option.

I've changed here the default to be multi-user for linux to be consistent. But I could just leave it as is with single-user install being the default which does not have this issue at all. The other option is to let the user pick at install time.

Also apparently this is intended behavior:

* https://github.com/NixOS/nix/issues/8248

Reply:
I have moved this change to a separate PR:

* https://github.com/status-im/status-mobile/pull/16357